build_tools/cmake/iree_bytecode_module.cmake

```
@@ -83,39 +83,42 @@ function(iree_bytecode_module)
   # Users can add their own additional directories as needed.
   list(APPEND _ARGS "--iree-hal-executable-object-search-path=\"${IREE_BINARY_DIR}\"")
 
-  # If an LLVM CPU backend is enabled, supply the linker tool.
-  if(IREE_LLD_TARGET)
-    set(_LINKER_TOOL_EXECUTABLE "$<TARGET_FILE:${IREE_LLD_TARGET}>")
-    list(APPEND _ARGS "--iree-llvmcpu-embedded-linker-path=\"${_LINKER_TOOL_EXECUTABLE}\"")
-    list(APPEND _ARGS "--iree-llvmcpu-wasm-linker-path=\"${_LINKER_TOOL_EXECUTABLE}\"")
-    # Note: --iree-llvmcpu-system-linker-path is left unspecified.
-  endif()
-
-  if(IREE_BYTECODE_MODULE_FORCE_LLVM_SYSTEM_LINKER)
-    list(APPEND _ARGS "--iree-llvmcpu-link-embedded=false")
-  endif()
-
-  if(IREE_BYTECODE_MODULE_ENABLE_ASAN)
-    list(APPEND _ARGS "--iree-llvmcpu-sanitize=address")
-  endif()
+  set(_OUTPUT_FILES "${_MODULE_FILE_NAME}")
 
-  # Support testing in TSan build dirs. Unlike other sanitizers, TSan is an
-  # ABI break: when the host code is built with TSan, the module must be too,
-  # otherwise we get crashes calling module code.
-  if(IREE_BYTECODE_MODULE_ENABLE_TSAN)
-    list(APPEND _ARGS "--iree-llvmcpu-sanitize=thread")
-  endif()
+  # This crude check is borrowed from iree_compile_flags_for_platform().
+  # This should be made more robust and shared in a common helper.
```

Can you keep an explanation comment too?
```suggestion
  # If the llvm-cpu backend is used, pass build-system-dependent flags.
  #
  # This crude check is borrowed from iree_compile_flags_for_platform().
  # This should be made more robust and shared in a common helper.
```

Calling the flags build-system-dependent (or using similar language) is important, because otherwise the flags could be added directly where the tests themselves are defined.

---

Added a comment two lines below where we actually pass build-system-dependent linker paths. This comment here was added about something else - "crude" refers to the expression `_RULE_FLAGS MATCHES "iree-hal-target-backends=llvm-cpu"`.

---

I'd still prefer my version of the comment there. Code is easier to scan when you can read it linearly, and "build-system-dependent" applies to _all_ of the flags in this section (including `--iree-llvmcpu-sanitize`, etc.)

```cmake
# If some condition is met, we need to add extra flags.
#
# We check for that condition with this mechanism:
# [explain specifics here]
if (_RULE_FLAGS MATCHES "iree-hal-target-backends=llvm-cpu")
  if (IREE_LLD_BINARY)
      list(APPEND _ARGS "--iree-llvmcpu-embedded-linker-path=\"${IREE_LLD_BINARY}\"")
```

compared to:
```cmake
# [explain specifics here]
if (_RULE_FLAGS MATCHES "iree-hal-target-backends=llvm-cpu")
  if (IREE_LLD_BINARY)
    # we need to add extra flags.
      list(APPEND _ARGS "--iree-llvmcpu-embedded-linker-path=\"${IREE_LLD_BINARY}\"")
...

---

> "build-system-dependent" applies to all of the flags in this section (including --iree-llvmcpu-sanitize, etc.)

That's what I didn't understand --- I though that by "build-system-dependent" you meant "strings that depend on expansion of CMake variables".

How about we just merge this so we get the byo-llvm CI coverage, and you do a follow-up with the improvements that you have in mind?

---

> > "build-system-dependent" applies to all of the flags in this section (including --iree-llvmcpu-sanitize, etc.)
> 
> That's what I didn't understand --- I though that by "build-system-dependent" you meant "strings that depend on expansion of CMake variables".

Ah, let me expand on that then...

If a leaf package like `runtime/src/iree/vm/test/` uses `--iree-llvmcpu-sanitize=address`, the targets defined there will only work if you happened to build with `IREE_ENABLE_ASAN`. Similar story for `--iree-llvmcpu-embedded-linker-path` with `IREE_LLD_TARGET`, `--iree-llvmcpu-target-triple` with cross compiling or building on Linux/Windows/macOS, etc.

So I wanted to preface this block of code with a comment like `# If the llvm-cpu backend is used, pass build-system-dependent flags.`, _then_ go into specifics about how configurations are checked and flags are automatically appended to the compile command.

> How about we just merge this so we get the byo-llvm CI coverage, and you do a follow-up with the improvements that you have in mind?

This is part of why I suggested splitting the PR into two. It's easier to review and merge the components separately.

---

Done